src/test/java/com/google/devtools/build/lib/skylark/SkylarkRuleClassFunctionsTest.java

```diff
@@ -1752,6 +1760,49 @@ public void testTargetsCanAddExecutionPlatformConstraints() throws Exception {
         .isEqualTo(ExecutionPlatformConstraintsAllowed.PER_TARGET);
   }
 
+  @Test
+  public void testTargetsCanAddExecutionPlatformConstraints_notEnabled() throws Exception {
+    StarlarkSemantics semantics =
+        StarlarkSemantics.DEFAULT_SEMANTICS.toBuilder()
+            .incompatibleDisallowRuleExecutionPlatformConstraintsAllowed(false)
+            .build();
+    ev = createEvaluationTestCase(semantics);
+    ev.initialize();
+
+    scratch.file("test/BUILD", "toolchain_type(name = 'my_toolchain_type')");
+    evalAndExport(
+        "def impl():",
+        "  return 0",
```

Comment:
Nit: AFAIK creating a target with these rules in these tests would fail with a different error anyway because you can't return an int from a rule implementation. `return []` instead or just `pass`

Same for the other new tests

Reply:
I was just copying registerDummyUserDefinedFunction, which I have switched back to using and also fixed.